plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2258,10 +2258,6 @@ private Optional<ConnectorTableExecuteHandle> getTableHandleForOptimize(Connecto
             throw new TrinoException(NOT_SUPPORTED, "OPTIMIZE procedure must be explicitly enabled via " + NON_TRANSACTIONAL_OPTIMIZE_ENABLED + " session property");
         }
 
-        if (retryMode != NO_RETRIES) {
```

Comment:
@losipiuk Is this intentional and something is missing to have it safely enabled? Or did it get disabled by chance?

Reply:
Probably not intentional. But enabling it would need a coverage by tests. Not sure if worthy given whole OPTIMIZE feature is feature-flagged and disbled by default.